### DISCUSSION TITLE
Connect to Ollama

### DISCUSSION THREAD

**Discussion by razvanab** [LostRuins/koboldcpp#1213] [Ideas]
I would like to use a llm model through ollama when using --nomodel with Koboltcpp.

**Comment by LostRuins** [comment]
You can click connect to external endpoint, and then connect it as a custom OpenAI API.
Then run ollama in openai api mode.

But koboldcpp should provide superior results. Why do you need ollama?

**Reply by LostRuins** [reply]
Edit: Ollama API is now supported!

**Comment by razvanab** [comment]
Ollama now supports downloading models directly from hugging faces. There is no need to download them separately in GGUF format for other apps.

**Comment by razvanab** [comment]
Yes, it did work changing the AI provider to OpenAI Compatible API.
Thank you.


Or not. It's asking for the API key to work, but Ollama doesn't use an API key.

**Reply by LostRuins** [reply]
Hi @razvanab you can just use any random string as the api key

**Reply by razvanab** [reply]
I tried. It didn't work.

**Comment by razvanab** [comment]
It is working now in the latest release. Great work, guys!
Hopefully you will manage to fix the streaming too in the next release.

👍

**Reply by LostRuins** [reply]
Glad it works well for you
